src/vp8.rs

```diff
@@ -1710,11 +1719,12 @@ impl<R: Read> Vp8Decoder<R> {
                 2
             };
 
-            if self.partitions[p].read_bool(128)? {
+            if reader.read_bool(128)? {
                 abs_value = -abs_value;
             }
 
-            block[ZIGZAG[i] as usize] =
+            // % 16 is added to eliminate a bounds check. Same as above.
+            block[(ZIGZAG[i] % 16) as usize] =
```

Comment:
I'm not very surprised there's little performance gain here and above, it's doing a mask instead of the conditional jump. [If only we could convince the compiler of the value range of values loaded form `ZIGZAG`](https://docs.rs/index-ext/latest/index_ext/tag/index.html#checked-constant-bounds) by encoding it in the type. But the available implementation, for such an `unsafe`-heavy interior, is a quite underdeveloped, I wouldn't recommend really using it for production yet even though having written it. And ergonomics are a nightmare.